### DISCUSSION TITLE
Is there any way to detect such audio noise when using aaudio

### DISCUSSION THREAD

**Discussion by xzhan96** [google/oboe#1931] [General]
Hi, many android devices has incompatibility problem with AAudio, such as silenced, noise, etc.
for example, please see this video(https://user-images.githubusercontent.com/13726257/209959288-2f8105ea-9215-427b-93df-0356637d6172.mp4), from 20th seconds, you can hear my sound with obvious noise.

I am thinking how to avoid such noise, one method is detect it in runtime and switch capture api to Java or OpenSL. Do you know is there a way or library can do that? I tried some but failed.

BTW, How to express this kind of noise more accurately in technical term? I use "electric current sound" before, but not sure its ok.

Thanks a lot.

**Comment by philburk** [comment]
The noise only happens when the voice is loud, not between words. It sounds to me like clipping.

But you recording level was very low so I would not expect clipping.
Some phone may clip when the volume is at maximum.

> one method is detect it in runtime and switch capture api to Java or OpenSL.

Do you have clipping when you record with Java or OpenSL ES?
You can tell OboeTester to record using OpenSL ES using the API menu.
Also try unchecking the MMAP box to see if Legacy AAudio works better.

You can use the SHARE button to save a WAV file recording to Drive or GMail.
You can then look at it in Audacity.

What phone are you using and what OS version?
